chain-factory/src/factory.rs

```diff
@@ -60,16 +64,13 @@ pub trait FactoryModule {
     ) {
         require!(!contracts_map.is_empty(), "Given contracts map is empty");
 
-        let mapped_array: ManagedVec<ManagedBuffer> = SC_ARRAY
-            .into_iter()
-            .map(|sc| ManagedBuffer::from(*sc))
-            .collect();
+        for contract in contracts_map {
+            require!(
+                contract.name != ScArray::None,
```

Comment:
1. Why do you need `ScArray::None`?
2. What happens if endpoint receives index 100?
3. If you use enum maybe use `contract.id` instead of `contract.name`

Reply:
The None is used so the index can start from 1, if it receives index 100 the transaction will fail.

Reply:
What is the problem if index 0 is a contract?

Reply:
The problem is that using index 0 in a storage mapper will not store anything inside the contract, and since the storage mapper `contracts_map` uses an index for the ID, it won't do anything for ID 0

Reply:
Actually, that's not true. An index of 0 will be nested encoded, so the storage key will be `contractsMap0`.

Reply:
My bad, I remove the `None`